src/rat/core/sarea/bot_filter.py

```diff
@@ -0,0 +1,236 @@
+import pandas as pd
+import numpy as np
+import os
+import geopandas as gpd
+from logging import getLogger
+from rat.utils.logging import LOG_NAME, LOG_LEVEL1_NAME
+
+log = getLogger(f"{LOG_NAME}.{__name__}")
+log_level1 = getLogger(f"{LOG_LEVEL1_NAME}.{__name__}")
+
+def bot_filter(tmsos_gee_path,basin_reservoir_shpfile_path,apply,bias_threshold,outlier_threshold,trend_threshold):
+    ''' 
+    Applies the BOT Filter to the surface area time series,
+    which filters the optical derived surface area with SAR surface area as the reference.
+    
+    Parameters
+    ----------
+    tmsos_gee_path : str
+        The path to the tmsos surface area folder within the basin data directory
+    basin_reservoir_shpfile_path : str
+        The path to the file containing reservoir shapefile data
+    apply: boolean
+        Toggles whether the BOT filter is run. It is also used to reset the BOT Filtered surface area values to TMSOS if required (set to False).
+        Values: True/False
+    bias_threshold: float
+        Controls the filtering intensity for the bias between Optical and SAR surface areas.
+        Values: 0 - 9
+    outlier_threshold: float
+        Controls the intensity of filtering out the outliers from the Optical surface area data.
+        Values: 0 - 9
+    trend_threshold: float
+        Controls the intensity of filtering the Optical surface area data based on the deviation in trends from the SAR surface area data.
+        Values: 0 - 9
+    '''
+   
+    #  Reading the reservoir shapefile data and creating the list of reservoirs in the basin. 
+    basin_reservoir_data = gpd.read_file(basin_reservoir_shpfile_path)
+    log_level1.debug(basin_reservoir_shpfile_path)
```

Comment:
Why debugging in loglevel1?

Reply:
Was added by mistake during testing. Removed